File: EVENTUAL.py
# !/usr/bin/python3

"""
https://www.codechef.com/COOK120B/problems/EVENTUAL
Even-tual Reduction.
"""


def solve(s, n):
    s = list(s)
    ans = ord(s[0])

    for i in range(1, n):
        ans ^= ord(s[i])

    if ans != 0: print("NO")
    else: print("YES")

    return


if __name__ == "__main__":
    t = int(input())

    while t:
        n = int(input())
        s = input()
        solve(s, n)
        t -= 1

File: EVENTUAL.cpp
/*
 * https://www.codechef.com/COOK120B/problems/EVENTUAL
 * Even-tual Reduction.
 */

# include<iostream>

void solve(std::string& s, int n)
{
    int ans = 0;

    for(int i = 0; i < n; i++)
    {
        ans ^= (int)s[i]-96;
    }
    std::cout<<ans<<std::endl;
    std::cout<<(ans ? "NO": "YES")<<std::endl;

    return;
}


int main()
{
    int t;
    std::cin>>t;

    while(t--)
    {
        int n;
        std::cin>>n;

        std::string s;
        std::cin>>s;
        solve(s, n);
    }
    return 0;
}
